### PR TITLE
feat(ui): search functionality

### PR DESCRIPTION
**Pull request by aniketkatkar97**
Search functionality added for the new docs. Currently, the search results are from older content for testing purposes. We will change that after we add new content.

- added search functionality

https://user-images.githubusercontent.com/51777795/211988860-836fa4bf-6a25-4896-b1e0-202ae1bb9f3d.mov

- loader will be shown while a search is in progress

https://user-images.githubusercontent.com/51777795/211988936-2f5c6439-c928-4afb-8a6c-d99b60a964d4.mov

- added global search shortcut (cmd/ctrl + k) to trigger a global search and hitting ESC will close the suggestions

https://user-images.githubusercontent.com/51777795/211989108-0157f109-3ca3-4a9f-b67f-74156b14b106.mov

- no data placeholder is shown in case no search results are found

https://user-images.githubusercontent.com/51777795/211989283-e69798ed-5f53-44b3-80a1-3e49f9faa78a.mov

### CODE REVIEW

File: .env.local [hunk 1]
```diff
@@ -0,0 +1,2 @@
+NEXT_PUBLIC_ALGOLIA_APP_ID=9JH19D2DQ0
+NEXT_PUBLIC_ALGOLIA_API_KEY=38758af91d7e29692db598a007069a29
```
Comment (chirag-madlani) [inline .env.local:2]: We should not commit any env variable revert it and force push it

File: package.json [hunk 2]
```diff
@@ -22,15 +22,17 @@
...
     "react-dom": "latest",
+    "react-instantsearch-hooks-web": "^6.38.2",
     "rehype-slug": "^5.0.1",
     "remark-unwrap-images": "^3.0.1",
     "slugify": "^1.6.5"
   },
   "devDependencies": {
+    "@types/lodash": "^4.14.191",
     "@types/node": "latest",
+    "@types/prismjs": "^1.26.0",
```
Comment (chirag-madlani) [inline package.json:33]: Do we need prismjs?

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (aniketkatkar97) [reply package.json:33]: Yes, it is in use

**Review by chirag-madlani — APPROVED** [review]
@harshach Can you also give it a look and share feedbacks if any?

**Pull request merged**